src/components/order/StatusLabel/index.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useMemo } from 'react'
+import styled from 'styled-components'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faCheckCircle, faClock, faDotCircle } from '@fortawesome/free-solid-svg-icons'
+
+import { OrderStatus } from 'api/operator'
+
+export type Props = { status: OrderStatus }
+
+const Wrapper = styled.div<Props>`
+  font-size: ${({ theme }): string => theme.fontSizeNormal};
+  font-weight: ${({ theme }): string => theme.fontWeightBold};
+  text-transform: capitalize;
+
+  border-radius: 0.4rem;
+
+  padding: 0.75em;
+  display: inline-block;
+
+  & > svg {
+    margin: 0 0.75rem 0 0;
+  }
+
+  ${({ theme, status }): string => {
+    let background, color
```

Comment:
cool, never though to do this :+1:

Reply:
man, i'm probably outdated. But I like CSS and bigger chunks of JS with some separation. why not do a pure static private function, and just use it here?

Reply:
No problem, can do that as well.
Would you rather this private function return css? Or only the color values?
Will have a PR for that shortly

Reply:
Follow up https://github.com/gnosis/gp-ui/pull/116